src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -2038,6 +2037,26 @@ private static void EnsureStatsAndDataIntegrity(DocumentsOperationContext contex
 
             Debug.Assert(first == stats.Start, $"Failed start check: {first} == {stats.Start}");
             Debug.Assert(last == stats.End, $"Failed end check: {last} == {stats.End}");
+
+            if (segment.NumberOfLiveEntries > 0)
+            {
+                foreach (var state in segment.SegmentValues.Span)
+                {
+                    Debug.Assert(double.IsNaN(state.First) == false, "First is NaN");
+                    Debug.Assert(double.IsNaN(state.Last) == false, "Last is NaN");
+                    Debug.Assert(double.IsNaN(state.Min) == false, "Min is NaN");
+                    Debug.Assert(double.IsNaN(state.Max) == false, "Max is NaN");
+                    Debug.Assert(double.IsNaN(state.Sum) == false, "Sum is NaN");
+                    Debug.Assert(double.IsNaN(state.Count) == false, "Count is NaN");
+                    Debug.Assert(state.Count > 0, "Count is Zero");
+                }
```

Comment:
Why do we have all of those asserts here? If we are validating this, let's move all of this to a `[Conditional("DEBUG")]` method.

Reply:
It is all under `[Conditional("DEBUG")]`